### DISCUSSION TITLE
How do Export 'ZData' From ODF?

### DISCUSSION THREAD

**Discussion by PedroPinh** [mtex-toolbox/mtex#1233] [Ask Anything]
After plotting an ODF, I would like to show the change in the fibers, according with the 'Z' data, like those examples:
![Fi42 NT editada](https://user-images.githubusercontent.com/91805626/135732681-7738411f-686d-4f0a-9a39-a0ff66fee727.jpg)
![WhatsApp Image 2021-09-30 at 12 42 36](https://user-images.githubusercontent.com/91805626/135732683-511e52cd-e96d-402c-bb5f-3d3f0491f86a.jpeg)
(These graphs aren't related with the presented ODF)

I tried to use:
t = findobj(gca,'Type','hggroup'); %(Create one variable 't')%
z=get(t,'Zdata'); %(Extract the Z values on a matrix)%

But it didn't work. Anyone has some tips?

**Comment by zmichels** [comment]
Does something like the end of this webpage help?

https://mtex-toolbox.github.io/ODFPlot.html

**Reply by PedroPinh** [reply]
It's not what we want. The command we want should export the data in matrix format, in order to manually choose the desired data. We don't know why the 'z' variable is created, but the 'Z' data (a.k.a. the intensity of ODF) aren't there. Must have some syntax or anything missing on the command...

**Comment by zmichels** [comment]
Sounds good… I thought the goal was the plots like you showed, which looked similar to me.

Best of luck.

**Reply by Cst1610123** [reply]
odf = calcODF(pf, 'resolution', 5 * degree);
odf.SS = specimenSymmetry('222');
 varargin={'phi2',[45]*degree};
    if odf.antipodal, ap = {'antipodal'}; else, ap = {}; end
    oS = newODFSectionPlot(odf.CS,odf.SS,ap{:},varargin{:});
     S3G = oS.makeGrid('resolution',5*degree,varargin{:});
    Z_origin = odf.eval(S3G);
